pkg/generator/caches.go

```diff
@@ -45,20 +45,54 @@ type Caches struct {
 	listeners           []*v2.Listener
 	statusVirtualHost   *route.VirtualHost
 	logger              *zap.SugaredLogger
+	ingressesToSync     map[string]struct{}
+	synced              chan bool
 }
 
-func NewCaches(logger *zap.SugaredLogger, kubernetesClient kubeclient.Interface, extAuthz bool) (*Caches, error) {
+func NewCaches(logger *zap.SugaredLogger, kubernetesClient kubeclient.Interface, extAuthz bool, ingressesToSync []*v1alpha1.Ingress) (*Caches, error) {
 	c := &Caches{
 		ingresses:           make(map[string]*v1alpha1.Ingress),
 		translatedIngresses: make(map[string]*translatedIngress),
 		clusters:            newClustersCache(logger.Named("cluster-cache")),
 		clustersToIngress:   make(map[string][]string),
 		logger:              logger,
+		synced:              make(chan bool),
 	}
 	err := c.initConfig(kubernetesClient, extAuthz)
+
+	// Listen for Sync events, triggered when there's a removal of an object of the "IngressesToSync" array
+	go c.syncEvents()
+
+	if ingressesToSync == nil {
+		// If ingressesToSync is empty, we send an event to the "synced" channel as we don't need to warm anything.
+		c.synced <- true
```

Comment:
Just close the channel here. The semantics are exactly the same as sending an event + closing it afterwards.

Reply:
I didn't know about that behavior! thanks, that's really useful! :)